### DISCUSSION TITLE
Performance Benchmark Testing (help wanted!)

### DISCUSSION THREAD

**Discussion by WaynePhillipsEA** [twinbasic/twinbasic#889] [Ideas]
As per the roadmap (#335), it is expected that I will be working on the optimizing compiler in Q3 of 2022.  With that in mind, I wanted to ask if there are members of the community here that would be interested in building a performance test suite that we can use to compare the tB compiler to VB6 (and VBA)? 

This would help me immensely as it means I would be able to concentrate on the tweaking of the performance, and less time writing tests.

I envisage that the performance test suite would cover all of the general language syntax, plus the built-in objects (Collection, Err, etc), and of course more thorough/complex algorithm tests that should help highlight the areas that most need attention.

I appreciate that some of you do not have VB6 in order to test compiled VB6 performance, but that shouldn't matter as there are plenty of us here that do have it, and can run the tests for you.

If you'd be willing to get involved, please drop a note in this discussion.  The plan would be to open a new repo on GitHub to contain this suite.

**Comment by FullValueRider** [comment]
One way of approaching this problem would be to get twinBasic on Rosetta Code.  

https://www.rosettacode.org/wiki/Category:Programming_Languages

On a related , and hopefully  upbeat note, I took the Binary tree benchmark from here

https://programming-language-benchmarks.vercel.app/problem/binarytrees

and had a bash (ho ho ho, groan) at C# and twinBasic versions. For the 18 and 15 inputs the timings in milliseconds I think I got were 140ms and 753ms for C# and  140ms and 1500ms for twinBasic.  This is on a Ryzen 3700X clocked at around 4.0 GHz.

The C# timings are comparable to those published for the 15 input, but about half the time for the 18 input.  Probably down to bigger cache compared to the Xeon(R) Platinum 8272CL CPU @ 2.60GHz (Model 85) used for the Website timings.

Unfortunately I don't have VB6.

**Comment by EduardoVB** [comment]
Maybe we could do something collaborative?
I think we first should define what is needed to do and how (or at least an approximation). It would be analysis/design.
Define some guidelines, so different developers can work, following the same rules.

I think this project is not big, but not small either (to do it right).
So, if different developers could work to add something as they have time, the project could be advanced.

It could be an interesting experience. It could also serve as a base for other collaborative projects (libraries, add-ins, documentation, test-units, whatever).
Talking about myself: I don't currently have experience in working in a group of developers for the same project.
